flink/v1.16/flink/src/main/java/org/apache/iceberg/flink/source/DataIterator.java

```diff
@@ -68,6 +69,23 @@ public DataIterator(
     this.recordOffset = 0L;
   }
 
+  public DataIterator(
```

Comment:
should we remove the other constructor (also to avoid code duplication)? this is an `@Internal` class. it should be ok.

Reply:
Done.